### PR TITLE
📃 feat: Linux Neptune (Valve's kernel for SteamDeck) installation

### LINKED ISSUE #269: Making it easier to setup up the Steam Deck OLED on vanilla arch.
**Is your feature request related to a problem? Please describe.**

Wi-Fi, Bluetooth as well as audio does not work on a base vanilla arch install for the Steam Deck OLED out of the box.

**Describe the solution you'd like**

A script to setup Wi-Fi, Bluetooth and the audio for the Steam Deck OLED on a vanilla arch install.

**Describe alternatives you've considered**

I've tried following the arch wiki with no luck after installing what it says to install to get the wi-fi etcetera working even after a restart.

Using something like Nobara which comes with steam deck Wi-Fi, Bluetooth and audio working out the box when using the Steam Deck version of Nobara.

Giving up and using SteamOS

**Additional context**
Edit: this might help as these were used in winesapos (which is based on arch linux) for the steam deck oled so a script to automatically insteall these would be great! You can find these being talked about on the winesapos github description where it talks about the steam deck being fully supported on winesapos.

**steam deck oled vanilla arch patches:**

https://pagure.io/kernel-fsync/blob/main/f/SOURCES/steamdeck-oled-audio.patch

https://aur.archlinux.org/packages/linux-firmware-valve

https://pagure.io/kernel-fsync/blob/main/f/SOURCES/steamdeck-oled-legion-go-bluetooth-hang.patch

https://pagure.io/kernel-fsync/blob/main/f/SOURCES/steamdeck-oled-wifi.patch

### PR DESCRIPTION
**Pull request by adamperkowski**
## Type of Change
- [x] New feature
- [x] Bug fix

## Description
Added a script that installs Linux Neptune and patched required for some components to work on Vanilla Arch. [ArchWiki](https://wiki.archlinux.org/title/Steam_Deck).

## Testing
~~Could not test on the deck so I am unable to confirm if Wi-Fi works properly, but~~ the script itself works.
https://github.com/ChrisTitusTech/linutil/issues/269#issuecomment-2380379049

About audio: https://github.com/ChrisTitusTech/linutil/issues/269#issuecomment-2381411675

## Issues / other PRs related
- Resolves #269

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

(📃 is a shell script btw)

### CODE REVIEW

**Comment by cartercanedy** [comment]
I think we should implement a confirmation prompt to prevent people from fat fingering on these kinds of significant changes

**Comment by cartercanedy** [comment]
I've done it before, and I've seen Chris wipe out his nvim cfg on stream, that would be big

**Comment by adamperkowski** [comment]
@cartercanedy On one hand, I agree, on the other one, Chris says he wants MINIMAL (none) user interaction in shell scripts. 
And we have **Important Actions** with **kernel modifications** + the script preview so I don't know about this to be honest.

**Comment by cartercanedy** [comment]
> Chris says he wants MINIMAL (none) user interaction in shell scripts

I was thinking about implementing this via a tui integration

**Comment by adamperkowski** [comment]
> I was thinking about implementing this via a tui integration

That might actually be a good idea for tui to automatically detect important actions (we could move these from strings to custom values in toml) and display a popup with confirmation. 
Open an issue, please.

File: core/tabs/system-setup/arch/linux-neptune.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        "$ESCALATION_TOOL" "$PACKAGER" -S --noconfirm linux-neptune linux-neptune-headers
+	"$ESCALATION_TOOL" mkinitcpio -P
+    else
+	printf "%b\n" "${GREEN}linux-neptune detected. Skipping installation.${RC}"
+    fi
+
+    if [ -f /etc/default/grub ]; then
+	printf "%b\n" "${CYAN}Updating GRUB...${RC}"
+	if ! grep -q '^UPDATEDEFAULT=' /etc/default/grub; then
+	    echo 'UPDATEDEFAULT=yes' | sudo tee -a /etc/default/grub
```
Comment (nnyyxxxx) [inline core/tabs/system-setup/arch/linux-neptune.sh:33]:
```suggestion
	    echo 'UPDATEDEFAULT=yes' | "$ESCALATION_TOOL" tee -a /etc/default/grub
```

File: core/tabs/system-setup/arch/linux-neptune.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	else
+   	    sudo sed -i 's/^UPDATEDEFAULT=.*/UPDATEDEFAULT=yes/' /etc/default/grub
```
Comment (nnyyxxxx) [inline core/tabs/system-setup/arch/linux-neptune.sh:35]:
```suggestion
   	    "$ESCALATION_TOOL" sed -i 's/^UPDATEDEFAULT=.*/UPDATEDEFAULT=yes/' /etc/default/grub
```

File: core/tabs/system-setup/arch/linux-neptune.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#! /bin/sh -e
```
Comment (nnyyxxxx) [inline core/tabs/system-setup/arch/linux-neptune.sh:1]:
```suggestion
#!/bin/sh -e
```
keep the consistency with other scripts

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

File: core/tabs/system-setup/arch/linux-neptune.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    "$ESCALATION_TOOL" cp /usr/lib/firmware/cs35l41-dsp1-spk-cali.wmfw  /usr/lib/firmware/cirrus/
+    "$ESCALATION_TOOL" cp /usr/lib/firmware/cs35l41-dsp1-spk-prot.bin  /usr/lib/firmware/cirrus/
+    "$ESCALATION_TOOL" cp /usr/lib/firmware/cs35l41-dsp1-spk-prot.wmfw  /usr/lib/firmware/cirrus/
+}
+
+checkEnv
+checkEscalationTool
+setUpRepos
+installKernel
+copyFirmwareFiles
```
Comment (nnyyxxxx) [inline core/tabs/system-setup/arch/linux-neptune.sh:60]: some of the indentation throughout the script looks f'ed up locally, might want to reformat all of this

**Review by nnyyxxxx — APPROVED** [review]

File: core/tabs/system-setup/arch/linux-neptune.sh [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    if [ -f /etc/default/grub ]; then
+        printf "%b\n" "${CYAN}Updating GRUB...${RC}"
+    if ! grep -q '^UPDATEDEFAULT=' /etc/default/grub; then
+        echo 'UPDATEDEFAULT=yes' | "$ESCALATION_TOOL" tee -a /etc/default/grub
+    else
+        "$ESCALATION_TOOL" sed -i 's/^UPDATEDEFAULT=.*/UPDATEDEFAULT=yes/' /etc/default/grub
+    fi
+    if [ -f /boot/grub/grub.cfg ]; then
+        "$ESCALATION_TOOL" grub-mkconfig -o /boot/grub/grub.cfg
+        else
```
Comment (nnyyxxxx) [inline core/tabs/system-setup/arch/linux-neptune.sh:39]: fix indentation on this ``else``

**Comment by adamperkowski** [comment]
I gotta fix my neovim with those whitespaces. Don't know what happened.

**Comment by adamperkowski** [comment]
@ChrisTitusTech read this: #269

**Review by ChrisTitusTech — CHANGES_REQUESTED** [review]
Add precondition that isolates this to only show on steamdeck

File: core/tabs/system-setup/tab_data.toml [hunk 6]
```diff
@@ -15,6 +15,12 @@ description = "This command installs a minimal arch server setup under 5 minutes
 script = "arch/server-setup.sh"
 task_list = "SI D"
 
+[[data.entries]]
+name ="Linux Neptune for SteamDeck"
+description = "Valve's fork of Linux Kernel for the SteamDeck"
+script = "arch/linux-neptune.sh"
+task_list = "I PFM K"
+
```
Comment (lj3954) [inline core/tabs/system-setup/tab_data.toml:23]:
```suggestion

[[data.preconditions]]
matches = false
data = { file = "/sys/devices/virtual/dmi/id/board_vendor" }
values = ["Valve", "Jupiter"]

```
source: https://www.reddit.com/r/SteamDeck/comments/telfda/how_to_detect_if_software_is_running_on_steamdeck/

**Review by lj3954 — CHANGES_REQUESTED** [review]

File: core/tabs/system-setup/tab_data.toml (hunk 6 above)
Comment (adamperkowski) [reply core/tabs/system-setup/tab_data.toml:23]: Okayy. Thank you. Will check with the issue's author to see if this works.

**Comment by adamperkowski** [comment]
Drafting this because there's a bug in `tab_data`

**Pull request merged**